plugin/src/test/java/org/opensearch/ml/processor/MLInferenceSearchResponseProcessorTests.java

```diff
@@ -978,14 +1055,18 @@ public void testProcessResponseCreateRewriteResponseListenerExceptionIgnoreFailu
 
         SearchResponse mockResponse = mock(SearchResponse.class);
         SearchHits searchHits = response.getHits();
+
+        InternalSearchResponse internalSearchResponse = new InternalSearchResponse(searchHits, null, null, null, false, null, 1);
+        when(mockResponse.getInternalResponse()).thenReturn(internalSearchResponse);
+
         RuntimeException mockException = new RuntimeException("Mock exception");
         AtomicInteger callCount = new AtomicInteger(0);
-        ;
+
         when(mockResponse.getHits()).thenAnswer(invocation -> {
 
             int count = callCount.getAndIncrement();
 
-            if (count == 2) {
+            if (count == 6) {
```

Comment:
Whats the significance of the number 6? If its tied to a object with a size perhaps we can use that instead? also maybe we can add a inequality since strict equality, I know you are using an AtomicInteger but the paranoid side of me feels that way 😅

Reply:
it was the 2nd time that it calls mockResponse.getHits() then throw exceptions, but now because I made the changes that calls 4 more times earlier, so it's calling the 6 time now to throw the exception at the tested function.